google/resource_compute_target_ssl_proxy_test.go

```diff
@@ -0,0 +1,189 @@
+	target := fmt.Sprintf("tssl-test-%s", acctest.RandString(10))
+	cert := fmt.Sprintf("tssl-test-%s", acctest.RandString(10))
+	backend := fmt.Sprintf("tssl-test-%s", acctest.RandString(10))
+	hc := fmt.Sprintf("tssl-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeTargetSslProxyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeTargetSslProxy_basic1(target, cert, backend, hc),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeTargetSslProxyExists(
+						"google_compute_target_ssl_proxy.foobar"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccComputeTargetSslProxy_update(t *testing.T) {
+	target := fmt.Sprintf("tssl-test-%s", acctest.RandString(10))
+	cert1 := fmt.Sprintf("tssl-test-%s", acctest.RandString(10))
+	cert2 := fmt.Sprintf("tssl-test-%s", acctest.RandString(10))
+	backend1 := fmt.Sprintf("tssl-test-%s", acctest.RandString(10))
+	backend2 := fmt.Sprintf("tssl-test-%s", acctest.RandString(10))
+	hc := fmt.Sprintf("tssl-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeTargetSslProxyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeTargetSslProxy_basic1(target, cert1, backend1, hc),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeTargetSslProxyExists(
+						"google_compute_target_ssl_proxy.foobar"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccComputeTargetSslProxy_basic2(target, cert1, cert2, backend1, backend2, hc),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeTargetSslProxyExists(
+						"google_compute_target_ssl_proxy.foobar"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckComputeTargetSslProxyDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_compute_target_ssl_proxy" {
+			continue
+		}
+
+		_, err := config.clientCompute.TargetSslProxies.Get(
+			config.Project, rs.Primary.ID).Do()
+		if err == nil {
+			return fmt.Errorf("TargetSslProxy still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckComputeTargetSslProxyExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		config := testAccProvider.Meta().(*Config)
+
+		found, err := config.clientCompute.TargetSslProxies.Get(
+			config.Project, rs.Primary.ID).Do()
+		if err != nil {
+			return err
+		}
+
+		if found.Name != rs.Primary.ID {
+			return fmt.Errorf("TargetSslProxy not found")
+		}
+
```

Comment:
Do we want to do more testing here (e.g. that the stored parameters match the set parameters)?

Reply:
Done